### PR TITLE
Option to support additional Service types

### PR DESCRIPTION
**Pull request by dbcelm**
Greetings..!!

I have added option to choose between ClusterIP, NodePort and LoadBalancer service types. This will come handy for variety of use cases:

e.g. when using aws-load-balancer controller with Cilium CNI default ClusterIP service type won't work and will give 504 when being reached from ingress as ALB Target Group won't be able to send health check traffic to ServiceIP as it will be Cilium managed and not from VPC CIDR range, hence user has to opt for target-type as instance and hence only Service Type NodePort will work in that case.

Also added test case for service template.

### CODE REVIEW

**Review by asdfgugus — COMMENTED** [review]
Thanks for the PR! 
I added some comments. Could a maintainer verify them? Thanks! 
In addition, please run the helm-docs command.

**Comment by dbcelm** [comment]
@asdfgugus pushed changes as per comments and also ran helm-docs

File: charts/opencost/templates/service.yaml [hunk 1]
```diff
@@ -14,7 +14,17 @@ metadata:
   {{- end }}
 spec:
   selector: {{- include "opencost.selectorLabels" . | nindent 4 }}
-  type: {{ .Values.service.type }}
+{{- if and .Values.service .Values.service.type }}
+  type: "{{ .Values.service.type }}"
+{{- else }}
+  type: ClusterIP
```
Comment (asdfgugus) [inline charts/opencost/templates/service.yaml:20]: Already default.

File: charts/opencost/templates/service.yaml [hunk 2]
```diff
@@ -14,7 +14,17 @@ metadata:
...
+{{- end }}
+{{- if (eq .Values.service.type "LoadBalancer") }}
+{{- if .Values.service.loadBalancerSourceRanges }}
+  loadBalancerSourceRanges:
+{{ toYaml .Values.service.loadBalancerSourceRanges | indent 4 }}
+  {{- end -}}
```
Comment (asdfgugus) [inline charts/opencost/templates/service.yaml:26]: Indentation does not match.

**Comment by dbcelm** [comment]
@asdfgugus made required edits

**Comment by asdfgugus** [comment]
Sorry, I meant to remove the two spaces. 
To finalize, could you merge the commits together for a better visibility?

**Comment by dbcelm** [comment]
> Sorry, I meant to remove the two spaces. To finalize, could you merge the commits together for a better visibility?
 if for "loadBalancerSourceRanges" then its desired indent with type:
 
 merged last 2 commits

**Comment by mattray** [comment]
This looks good to me, could you bump the chart version to 1.36.0?

**Comment by dbcelm** [comment]
> This looks good to me, could you bump the chart version to 1.36.0?

bumped chart version to 1.36.0

**Comment by mattray** [comment]
@dbcelm sorry, there's been another release. Could you move it to 1.37.0?

**Comment by dbcelm** [comment]
> @dbcelm sorry, there's been another release. Could you move it to 1.37.0?

done

**Review by mattray — APPROVED** [review]

**Pull request merged**